resource-management/test/resourceRegionMgrSimulator/data/regionNodeEvents_test.go

```diff
@@ -78,13 +78,51 @@ func TestGetRegionNodeModifiedEventsCRV(t *testing.T) {
 	// 3.987µs
 	duration = time.Since(start)
 	assert.NotNil(t, modifiedEvents)
-	assert.Equal(t, 10, len(modifiedEvents))
-	t.Logf("Time to get %d update events: %v", count, duration)
+	assert.Equal(t, rpNum, len(modifiedEvents))
+	t.Logf("Time to get %d update events in Daily data pattern: %v", count, duration)
 	assert.Equal(t, atEachMin10*2, int(count))
 
 	//check remaining event list
-	assert.Equal(t, 10, len(RegionNodeUpdateEventList))
-	for i := 0; i < 10; i++ {
+	assert.Equal(t, rpNum, len(RegionNodeUpdateEventList))
+	for i := 0; i < rpNum; i++ {
 		assert.Nil(t, nil, RegionNodeUpdateEventList[i])
 	}
+
+	// generate update node events of Outage pattern
+	makeOneRPDown()
+
+	// get update nodes
+	for j := 0; j < location.GetRPNum(); j++ {
+		rvLoc := types.RvLocation{
+			Region:    location.Region(RegionId),
+			Partition: location.ResourcePartition(j),
+		}
+		rvs[rvLoc] = uint64(nodesPerRP)
+	}
+	start = time.Now()
+	modifiedEvents, count = GetRegionNodeModifiedEventsCRV(rvs)
+	// 3.929264ms
+	duration = time.Since(start)
+	assert.NotNil(t, modifiedEvents)
+	assert.Equal(t, rpNum, len(modifiedEvents))
+	t.Logf("Time to get %d update events in Outage data pattern: %v", count, duration)
+	assert.Equal(t, uint64(nodesPerRP), count)
+
+	// generate update node events of Outage pattern again
```

Comment:
I am confused here, what's the purpose of two separate call to makeOneRPDown and have the same code duplicated? To verify node events being depleted after GetRegionNodeModifiedEventsCRV?

Why line 113-119 needs to be done again?

Reply:
I only test two times and I can remove the codes for second time.